File: Tutorials/C#/Tutorials/Tutorials/Tutorial11.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Text;
using System.Windows.Forms;
using TutorialsBase;

namespace TutorialsCS4
{
    public partial class Tutorial11 : ITutorial
    {
        IHost _hostApplication;

        #region ITutorial Member

        public void Run()
        {
            string message = _hostApplication.LCID == 1033 ? "This tutorial doens't contain example code" : "Dieses Tutorial enthält keinen Beispielcode";
            MessageBox.Show(message, "Tutorial11", MessageBoxButtons.OK, MessageBoxIcon.Information);
        }

        public void Connect(IHost hostApplication)
        {
            _hostApplication = hostApplication;
        }

        public void Disconnect()
        {

        }

        public void ChangeLanguage(int lcid)
        {

        }

        public string Uri
        {
            get { return _hostApplication.LCID == 1033 ? "http://netoffice.codeplex.com/wikipage?title=Tutorial11_EN_CS" : "http://netoffice.codeplex.com/wikipage?title=Tutorial11_DE_CS"; }
        }

        public string Caption
        {
            get { return "Tutorial11"; }
        }


        public string Description
        {
            get { return _hostApplication.LCID == 1033 ? "Addin Deployment" : "Addins auf anderen System installieren"; }
        }

        public UserControl Panel
        {
            get { return null; }
        }

        #endregion
    }
}

File: Tutorials/VB/Tutorials/Tutorials/Tutorial11.vb
Public Class Tutorial11
    Implements ITutorial

    Dim _hostApplication As IHost

#Region "ITutorial Member"

    Public Sub Run() Implements TutorialsBase.ITutorial.Run

        Dim message As String = IIf(_hostApplication.LCID = 1033, "This tutorial doens't contain example code", "Dieses Tutorial enthält keinen Beispielcode")
        MessageBox.Show(message, "Tutorial11", MessageBoxButtons.OK, MessageBoxIcon.Information)

    End Sub

    Public ReadOnly Property Caption As String Implements TutorialsBase.ITutorial.Caption
        Get
            Return "Tutorial11"
        End Get
    End Property

    Public ReadOnly Property Description As String Implements TutorialsBase.ITutorial.Description
        Get
            Return IIf(_hostApplication.LCID = 1033, "Addin Deployment", "Addins auf anderen System installieren")
        End Get
    End Property

    Public Sub Connect(ByVal hostApplication As TutorialsBase.IHost) Implements TutorialsBase.ITutorial.Connect

        _hostApplication = hostApplication

    End Sub

    Public Sub ChangeLanguage(ByVal lcid As Integer) Implements TutorialsBase.ITutorial.ChangeLanguage

    End Sub

    Public Sub Disconnect() Implements TutorialsBase.ITutorial.Disconnect

    End Sub

    Public ReadOnly Property Panel As System.Windows.Forms.UserControl Implements TutorialsBase.ITutorial.Panel
        Get
            Return Nothing
        End Get
    End Property


    Public ReadOnly Property Uri As String Implements TutorialsBase.ITutorial.Uri
        Get
            Return IIf(_hostApplication.LCID = 1033, "http://netoffice.codeplex.com/wikipage?title=Tutorial11_EN_VB", "http://netoffice.codeplex.com/wikipage?title=Tutorial11_DE_VB")
        End Get
    End Property

#End Region

End Class